### ISSUE TITLE
Model collapse when training with GAN + Adabelief (no Learning Rate Dropout) and Gradient Clipping

### ISSUE THREAD

**Issue by ThomasBardem** [iperov/DeepFaceLab#5239]
Using the latest release, training a pretrained DF-UD 320 model with dims 320/72/72/16 on bunch of random faces before actual training.

Model has AB enabled from the start (both pretraining and training done with it enabled). I've ran during the training set most options, starting with Random Warp, then no RW, Uniform Yaw, bit of true face, mouth and eye priority until I've enabled learning rate dropout for a while and then disabled it and enabled GAN at 0.1 power with default patch size and dims.

Model trained fine for some time until it collapsed, I've decided to enabled gradient clipping despite reading up about some potential issues with it running on models using adabelief optimizer but still it also collapsed after some time producing results like this: https://imgur.com/a/fHDQc4v

When gradient clipping was disabled the collapsed face previews were completely black with DST faces looking fine, with gradient clipping enabled I'm getting these black/blue swirls and DST faces are all distorted and just look bad.

In both cases loss values go up to 4.100 like values. This happened also when the GAN values for patch size and dims were changed (lower than default).

Rest of the setup: updated Windows 10, AVX supported CPU, 1080Ti, Win 10 fix applied (GPU scheduling one)

This is really frustrating as I don't see any obvious reasons for why this model would collapse 3 times already, old GAN never caused any such issues and I was also able to turn on and off gradient clipping with no issues too, now with AB turning gradient clipping causes issues and training with GAN collapses model after just 1-3 hours of training no matter if gradient clipping is enabled or not.

Other thing I've noticed too is that models sometimes seem to train much faster, before AB models would always run at the same speed and use up same amount of VRAM, now with AB enabled models VRAM requirements for the same model can differ and sometimes the model may use 1-1,5GB more or less, which also affects training speed which when working correctly is under 800ms and when it's consuming more for some reason speed also goes down with iterations reaching 3000ms with usual 16th iteration raises going as high as 8000ms. I always make sure the available VRAM is the same before running the model and doesn't fully load the cards VRAM (usage never goes above 10,6GB on 11GB card and minimum usage when just the display is being driven by the GPU ranges between 0.4-0.8GB before launching the training.

**Comment by iperov** [comment]
read the help for the GAN option

**Comment by ThomasBardem** [comment]
I've read the GAN option, I know how to train GAN, the issue is that something is broken.

Here is what the GAN results look like shortly before model starts to collapse: https://i.imgur.com/nKMCv9d.png

Those red dots, it's the same thing I've seen when enabling/disabling gradient clipping on a model that had adabelief enabled.

To confirm my case I've trained a model which had gradient clipping enabled from the start and never had LRD enabled, GAN training has been going for 12 hours with no issues, model has the same parameters and was trained on the same dataset, absolutetly everything about how it was trained is exactly the same (minus eye mouth priority but I doubt it could be the case).

**Comment by iperov** [comment]
Offer a fix. Or dont use gan
